ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java

```diff
@@ -289,11 +293,15 @@ public Result<PCollection<SponsoredInteraction>, PubsubMessage> expand(
           if (SponsoredInteraction.FORM_DESKTOP.equals(interaction.getFormFactor())
               && SponsoredInteraction.SOURCE_SUGGEST.equals(interaction.getSource())) {
 
-            String customDataParam = builtUrl.getQueryParam(BuildReportingUrl.PARAM_CUSTOM_DATA);
-            builtUrl.addQueryParam(BuildReportingUrl.PARAM_CUSTOM_DATA,
-                Optional.ofNullable(interaction.getScenario())
-                    .map((pref) -> String.format("%s_%s", customDataParam, pref))
-                    .orElse(customDataParam));
+            Stream<Optional<String>> customDataElements = Stream.of(
+                Optional.ofNullable(interaction.getScenario()),
+                Optional.ofNullable(interaction.getMatchType()));
+
+            String originalValue = builtUrl.getQueryParam(BuildReportingUrl.PARAM_CUSTOM_DATA);
+            String customDataParam = customDataElements.filter(opt -> !opt.isEmpty())
```

Comment:
This should use the built-in `isPresent` instead of inverting `isEmpty`
```suggestion
            String customDataParam = customDataElements.filter(Optional::isPresent)
```